backend/app/services/session_service.py

```diff
@@ -102,6 +103,29 @@ def fetch_paginated_sessions(
     def create_new_session(
         self, session_data: SessionCreate, user_profile: UserProfile
     ) -> SessionRead:
+        # Enforce daily session limit for non-admin users
+        if user_profile.account_role != AccountRole.admin:
+            # Get session limit from AppConfig
+            session_limit_config = self.db.exec(
+                select(AppConfig.value).where(AppConfig.key == 'dailyUserSessionLimit')
+            ).first()
+            if session_limit_config is not None:
+                session_limit = int(session_limit_config)
+                # Count user's sessions started today
+                today = datetime.now(UTC).date()
+                scenario_ids = self._get_user_scenario_ids(user_profile.id)
+                if scenario_ids:
+                    session_count = self.db.exec(
+                        select(func.count())
+                        .select_from(Session)
+                        .where(Session.scenario_id.in_(scenario_ids))
+                        .where(func.date(Session.started_at) == today)
+                    ).one()
```

Comment:
You can't get the session count from the session table because users have the ability to delete sessions and therefore bypass this daily session limit. We need to have a session counter in the database that we increment with each new session. More details are in [the issue](https://github.com/jst-seminar-rostlab-tum/personio-foundation-coachai/issues/554).